docs/src/routes.jsx

```diff
@@ -11,495 +16,39 @@ const routes = (
-                        import('./templates/pageDataGroups')
-                            .then((module) => callback(null, module.default));
-                    }}
-                    path="data-groups"
-                />
-                <Route
-                    getComponent={(location, callback) => {
-                        import('./templates/pageDetailsWindow')
-                            .then((module) => callback(null, module.default));
-                    }}
-                    path="details-window"
-                />
-                <Route
-                    getComponent={(location, callback) => {
-                        import('./templates/pageFiltersDrawer')
-                            .then((module) => callback(null, module.default));
-                    }}
-                    path="filters-drawer"
-                />
-                <Route
-                    getComponent={(location, callback) => {
-                        import('./templates/pageFiltersRail')
-                            .then((module) => callback(null, module.default));
-                    }}
-                    path="filters-rail"
-                />
-                <Route
-                    getComponent={(location, callback) => {
-                        import('./templates/pageDemo')
-                            .then((module) => callback(null, module.default));
-                    }}
-                    path="demo"
-                />
+        {map(navigationItems, (levelOne) => (
+            <Route
+                key={`${levelOne.path}`}
+                path={`${levelOne.path}`}
+            >
+                {!isEmpty(levelOne.subSections) ? map(levelOne.subSections, (levelTwo) => (
+                    <Route
+                        key={`${levelTwo.path}`}
+                        path={`${levelTwo.path}`}
+                    >
+                        {map(levelTwo.items, (levelTwoItem) => (
+                            <Route
+                                getComponent={(location, callback) => {
+                                    import(`./${levelTwo.directory}/${levelTwoItem.component}`)
+                                        .then((module) => callback(null, module.default));
+                                }}
+                                key={`${levelTwoItem.path}`}
+                                path={levelTwoItem.path}
+                            />
+                        ))}
+                    </Route>
+                )) : map(levelOne.items, (levelOneItem) => (
+                    <Route
+                        getComponent={(location, callback) => {
+                            import(`./${levelOne.directory}/${levelOneItem.component}`)
+                                .then((module) => callback(null, module.default));
+                        }}
+                        key={`${levelOneItem.path}`}
+                        path={levelOneItem.path}
+                    />
+                ))}
```

Comment:
Nice and clean!  I like!

Reply:
Yeah, I believe this will help with maintenance.